test/parallel/test-zlib-from-gzip-with-trailing-garbage.js

```diff
@@ -37,11 +37,10 @@ assert.throws(
   /^Error: unknown compression method$/
 );
 
-zlib.gunzip(data, common.mustCall((err, result) => {
-  assert(err instanceof Error);
-  assert.strictEqual(err.code, 'Z_DATA_ERROR');
-  assert.strictEqual(err.message, 'unknown compression method');
-  assert.strictEqual(result, undefined);
+zlib.gunzip(data, common.expectsError({
+  code: 'Z_DATA_ERROR',
+  type: Error,
+  message: 'unknown compression method'
 }));
```

Comment:
This loses the check that `result` is `undefined`. I'd prefer leaving the old code. I don't think `common.expectsError()` adds significant value, but the check for `result` is in fact significant.